service/s3/internal/customizations/process_arn_resource.go

```diff
@@ -250,8 +251,15 @@ func buildAccessPointRequest(ctx context.Context, options accesspointOptions) (c
 
 	resolveService := tv.Service
 
+	eo := options.EndpointResolverOptions
+	if options.UseDualstack {
+		eo.UseDualStack = internalendpoints.EnableDualStackEndpoint
+	} else {
+		eo.UseDualStack = internalendpoints.DisableDualStackEndpoint
```

Comment:
why disable instead of unset? what's the difference in unset vs disable?

Reply:
`Unset` effectively maps to disabled, but `Unset` serves the purpose to allow us to see if the option was explicitly distinguish whether the option was set by the user rather then using a pointer.